src/arrayMethodJoin.js

```diff
@@ -5,7 +5,28 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let internalSeparator;
+    let resultString = '';
+
+    if (separator === undefined) {
+      internalSeparator = ',';
+    } else {
+      internalSeparator = separator;
+    }
```

Comment:
add a default value for argument separator 

`[].__proto__.join2 = function(separator = ',') {`